docs/client_retries.md

```diff
@@ -0,0 +1,267 @@
+              .setTotalTimeout(Duration.ofMillis(60000L))
+              .build();
+```
+
+The default configurations are mapping [here](https://github.com/googleapis/google-cloud-java/blob/d9da511b4b56302e509abe8b2d919a15ea7dcae7/java-asset/google-cloud-asset/src/main/java/com/google/cloud/asset/v1/stub/AssetServiceStubSettings.java#L1306-L1474)
+
+Example:
+```java
+      builder
+          .exportAssetsSettings()
+          .setRetryableCodes(RETRYABLE_CODE_DEFINITIONS.get("no_retry_0_codes"))
+          .setRetrySettings(RETRY_PARAM_DEFINITIONS.get("no_retry_0_params"));
+```
+
+## Google Cloud Client Library Retry Concepts
+Enabling retries allow an RPC multiple attempts to try and achieve a successful call. A successful call
+is a response from a server that returns an `OK` Status Code (from gRPC) or a `2xx` Status Code (from HttpJson).
+
+### Configurable Retry Params
+- Retry Status Code: Set of failure status codes to retry on
+- Retry Time/ Attempt Bounds: Configurable retry settings (via [RetrySettings](https://cloud.google.com/java/docs/reference/gax/latest/com.google.api.gax.retrying.RetrySettings) class) to define the retry bounds
+
+### When is an RPC retried
+Take a sample RetrySettings configuration
+```java
+settings =
+  RetrySettings.newBuilder()
+      .setInitialRetryDelay(Duration.ofMillis(100L))
+      .setRetryDelayMultiplier(1.3)
+      .setMaxRetryDelay(Duration.ofMillis(60000L))
+      .setInitialRpcTimeout(Duration.ofMillis(60000L))
+      .setRpcTimeoutMultiplier(1.0)
+      .setMaxRpcTimeout(Duration.ofMillis(60000L))
+      .setTotalTimeout(Duration.ofMillis(60000L))
+      .build();
+```
+Individual RPC Bounds (an attempt) are controlled by the following settings:
+- setInitialRetryDelay
+- setRetryDelayMultiplier
+- setMaxRetryDelay
+- setInitialRpcTimeout
+- setRpcTimeoutMultiplier
+- setMaxRpcTimeout
+
+Total RPC Bounds are controlled by the following settings:
+- setTotalTimeout
+- setAttemptCount
+
+An RPC will be retried when _both_ of the following scenarios occur:
+- Non-successful status code is received by the library and the status code is marked as retryable*
+- An RPC invocation exceeds the individual RPC bounds, but still falls within total RPC bounds**
+
+Note: If only one (or neither) of the scenarios above are true, then the RPC will not be retried.
+i.e. If the total timeout has not been exceeded, but the latest attempt receives a non-retryable status code.
+
+*The client library will mark a status code as retryable internally. It is marked as retryable if the response's
+status code matches with any of an RPC's configured retryable status codes.
+
+**When configuring the RPC bounds, you may configure the bounds for each attempt as well as the
+total RPC's bounds. The retry algorithm will ensure that an individual attempt's bounds falls within
+the total RPC's bounds.
+
+### Jitter
+Jitter is added variance via randomness to spread out when the RPCs are invoked. By default, Google Cloud
```

Comment:
Is there an official link or something to what `Jitter` is?

Reply:
We don't have an official guide of jitter. I can look for a more public guide regarding this, but I don't think google has one.

Reply:
Okay sorry I think the capitalization threw me off; I thought it was like an official library or something. But seems like its just a general software term.